src/client/gui/lib/update_available.dart

```diff
@@ -15,14 +18,38 @@ class UpdateNotifier extends Notifier<UpdateInfo> {
 
   void set(UpdateInfo updateInfo) {
     if (updateInfo.version.isBlank) return;
+
     final updateNotificationExists = ref.read(notificationsProvider).any((n) {
       return n is UpdateAvailableNotification && n.updateInfo == updateInfo;
     });
     if (updateNotificationExists) return;
+
+    // In-app notification
     ref
         .read(notificationsProvider.notifier)
         .add(UpdateAvailableNotification(updateInfo));
+
+    // Update the state
     state = updateInfo;
+
+    // Create and show a local notification
+    _showLocalNotification(updateInfo);
+  }
+
+  void _showLocalNotification(UpdateInfo updateInfo) {
+    if (!mpPlatform.showLocalUpdateNotifications) {
+      // Do not show notification on Linux
```

Comment:
I think you can remove this comment now, since it is no longer the responsibility of this method to decide on which platform the notification is shown. Then you can one-line `if (...) return;`